src/DevBetterWeb.Infrastructure/Handlers/MemberAddressUpdatedHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using DevBetterWeb.Core.Events;
+using DevBetterWeb.Core.Interfaces;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+
+namespace DevBetterWeb.Infrastructure.Handlers
+{
+  class MemberAddressUpdatedHandler : IHandle<MemberAddressUpdatedEvent>
+  {
+    public MemberAddressUpdatedHandler(IConfiguration configuration)
+    {
+      _configuration = configuration;
+    }
+
+    public IConfiguration _configuration { get; }
+
+    public async Task Handle(MemberAddressUpdatedEvent addressUpdatedEvent)
+    {
+      var member = addressUpdatedEvent.Member;
+      var addressParts = member.Address?.Split(',');
+
+      if (addressParts is not null)
+      {
+        var cityDetailsAPIResponse = JObject.Parse(await GetMapCoordinates(addressParts[1]));
+        var latResponse = cityDetailsAPIResponse.SelectToken("results[0].geometry.location.lat");
+        var lngResponse = cityDetailsAPIResponse.SelectToken("results[0].geometry.location.lng");
+
+        member.CityLatitude = latResponse?.ToObject<decimal>();
+        member.CityLongitude = lngResponse?.ToObject<decimal>();
+      }
```

Comment:
I'm guessing the reason you moved the code in the `SaveChanges` method is to make sure this handler gets called before changes are saved, so that the CityLatitude and CityLongitude are also saved. Your approach works, but it's not terribly clean. We should think about a better approach, such as implementing an immediate domain event pattern (which has its own complexities, to be sure). Another option would be to have some totally out-of-proc service be responsible for the lat/long updates, and this handler would just put a command to do the check on a queue which the other process would handle, but that's obviously more complex, too. Let's discuss - I'm leaning toward adding immediate domain events to the solution and using them for this.